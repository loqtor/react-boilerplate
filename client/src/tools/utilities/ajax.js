import axios from 'axios';

const ENVIRONMENT_TO_HOST = {
  baseUrl: process.env.REACT_APP_API_BASE_URL,
};

const ABSOLUTE_URL_REGEX = /^https?:\/\/|^\/\//i;

const isRequestSuccessful = (response) => {
  const statusFirstDigit = response.request.status.toString().charAt(0);
  return statusFirstDigit !== '4' && statusFirstDigit !== '5';
};

const processParams = (url, params) => {
  const baseUrl = `${url}?`;

  return Object.entries(params).reduce((accum, param) => {
    if (accum !== baseUrl) {
      return `${accum}&${param[0]}=${param[1]}`;
    }

    return `${accum}${param[0]}=${param[1]}`;
  }, baseUrl);
};

export async function makeRequest(uri, method, headers, data, params = undefined) {
  const requestMethod = method || 'get';
  const defaultHeaders = {
    'Accept': 'application/json',
    'Content-Type': 'application/json',
  };

  const { baseUrl } = ENVIRONMENT_TO_HOST;
  const requestUrl = ABSOLUTE_URL_REGEX.test(uri) ? uri : `${baseUrl || ''}${uri}`;

  const hasParams = params !== undefined;

  /** Process params object and return concatenated base URL and any params */
  const processedUrl = hasParams ? processParams(requestUrl, params) : requestUrl;

  const response = await axios({
    url: processedUrl,
    method: requestMethod,
    headers: {
      ...defaultHeaders,
      ...headers,
    },
    data,
  });

  if (isRequestSuccessful(response)) {
    return response.data;
  }

  throw response.response;
}

export default makeRequest;

export async function fetchJSON(url) {
  const response = await axios({
    url,
    method: 'get',
  });

  if (isRequestSuccessful(response)) {
    return response.data;
  }

  throw response.response;
}
