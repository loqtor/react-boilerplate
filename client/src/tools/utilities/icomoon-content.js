// NOTE: This is a generated file, nothing you do here matters
// The source is the icomoon file located at 'public/assets/icons/selection.json'
// The script that generates this file is located at tools/icomoon-generator.js
// To update this file, selection.json must be changed and
// the script run 'yarn run generate-icons'
import React from 'react';

export default {
  'preview': <span className="icon icon-preview" />,
  'check': <span className="icon icon-check" />,
  'add': <span className="icon icon-add" />,
  'search': <span className="icon icon-search" />,
  'dropdown': <span className="icon icon-dropdown" />,
  'chevron-down': <span className="icon icon-chevron-down" />,
  'chevron-up': <span className="icon icon-chevron-up" />,
  'chevron-left': <span className="icon icon-chevron-left" />,
  'chevron-right': <span className="icon icon-chevron-right" />,
  'arrow-back': <span className="icon icon-arrow-back" />,
  '10-back': <span className="icon icon-10-back"><span className="path1" key="1" /><span className="path2" key="2" /><span className="path3" key="3" /></span>,
  'play': <span className="icon icon-play" />,
  'player-play': <span className="icon icon-player-play" />,
  'player-pause': <span className="icon icon-player-pause"><span className="path1" key="1" /><span className="path2" key="2" /></span>,
  '10-forward': <span className="icon icon-10-forward"><span className="path1" key="1" /><span className="path2" key="2" /><span className="path3" key="3" /></span>,
  'captions': <span className="icon icon-captions" />,
  'captions-active': <span className="icon icon-captions-active"><span className="path1" key="1" /><span className="path2" key="2" /></span>,
  'episodes': <span className="icon icon-episodes"><span className="path1" key="1" /><span className="path2" key="2" /></span>,
  'full-screen-enter': <span className="icon icon-full-screen-enter" />,
  'full-screen-exit': <span className="icon icon-full-screen-exit" />,
  'skip': <span className="icon icon-skip" />,
  'mute': <span className="icon icon-mute" />,
  'volume': <span className="icon icon-volume" />,
};
