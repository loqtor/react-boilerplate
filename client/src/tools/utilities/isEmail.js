/**
 * Validates whether a given string is an email address.
 *
 * @param {String} string An input string to be validated
 */
const EMAIL_REGEX = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

const isEmail = string => EMAIL_REGEX.test(string);

export default isEmail;
