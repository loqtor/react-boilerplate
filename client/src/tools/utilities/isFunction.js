/**
 * Tests whether variable is a function
 *
 * @param {functionToCheck} any any variable to see if it's a function
 */
const isFunction = functionToCheck => functionToCheck && {}.toString.call(functionToCheck) === '[object Function]';

export default isFunction;
