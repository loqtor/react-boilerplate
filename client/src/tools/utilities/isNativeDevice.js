/**
 * Returns true or false if userAgent matches regex
 *
 */
export const isNativeDevice = /Android|webOS|iPhone|iPad|iPod|Opera Mini/i.test(
  navigator.userAgent,
);
