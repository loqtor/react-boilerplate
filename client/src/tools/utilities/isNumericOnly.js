/**
* Regex to test if a string contains only numeric
* characters
*
* @param {String} string An input string to be validated
*/
const isNumericOnly = string => /^[0-9\b]+$/.test(string);

export default isNumericOnly;
