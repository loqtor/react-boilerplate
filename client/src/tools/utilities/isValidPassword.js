/**
* Regex to test if a string has at least one
* uppercase character and one
* numeric character
*
* @param {String} string An input string to be validated
*/
const isValidPassword = string => /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\s).{8,}$/.test(string);

export default isValidPassword;
