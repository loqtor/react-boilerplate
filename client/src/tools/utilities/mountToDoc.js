/*
 * Utility to provide aXe with a fake DOM to allow for accessibility testing
 * on mounted components within our Jest tests
*/
import { mountWithIntl } from './enzymeIntlTestHelper';

let wrapper;

export default function mountToDoc(component) {
  const container = mountWithIntl(component);

  if (!wrapper) {
    wrapper = document.createElement('main');
    document.body.appendChild(wrapper);
  }

  wrapper.innerHTML = '';
  wrapper.appendChild(container.getDOMNode());

  return container;
}
