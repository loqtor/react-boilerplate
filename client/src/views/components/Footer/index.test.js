import React from 'react';
import axe from 'axe-core';
import toJson from 'enzyme-to-json';
import { injectIntl } from 'react-intl';

import mountToDoc from '../../../tools/utilities/mountToDoc';
import { shallowWithIntl } from '../../../tools/utilities/enzymeIntlTestHelper';

import { Footer } from './index';

const FooterWithIntl = injectIntl(Footer);

const setup = (render, props) => {
  const defaultProps = {
  };

  const component = render(<FooterWithIntl {...defaultProps} {...props} />);

  return {
    actual: component,
  };
};

describe('index.test.jsx', () => {
  it('renders correctly', () => {
    const { actual } = setup(shallowWithIntl, {});
    expect(toJson(actual)).toMatchSnapshot();
  });

  it('has no accessibility violations', (done) => {
    const { actual } = setup(mountToDoc, {});
    const componentNode = actual.getDOMNode();

    axe.run(componentNode, (err, { violations }) => {
      expect(err).toBe(null);
      expect(violations).toHaveLength(0);
      done();
    });
  });
});
