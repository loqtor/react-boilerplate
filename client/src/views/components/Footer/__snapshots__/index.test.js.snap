// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.test.jsx renders correctly 1`] = `
<Footer
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en-NZ",
      "messages": Object {
        "404": "The link may be broken or the page may have been removed. ",
        "APP_TITLE": "**APP_TITLE**",
        "BACK": "Back",
        "BY": "By ",
        "CANCEL": "Cancel",
        "CHANGE": "Change",
        "CLEAR_INPUT": "Clear input. ",
        "CLOSED": "Closed",
        "CONTINUE": "Continue",
        "DATE": "Date",
        "DATE_OF_BIRTH": "Date of birth",
        "DAY": "Day",
        "DELETE": "Delete",
        "DESCRIPTION": "Description",
        "DETAILS": "Details",
        "DONE": "Done",
        "EDIT": "Edit",
        "EMAIL": "Email",
        "FACEBOOK": "Facebook",
        "FINISH": "Finish",
        "FOOTER_DISCLAIMER": "**APP_TITLE** Â© 2018 All Rights Reserved",
        "FROM": "From",
        "FULL_NAME": "Full name",
        "GOT_IT": "Got it!",
        "GO_TO_LOGIN": "Go to login",
        "GREETING": "Hi ",
        "HOME": "Home",
        "IS": "Is ",
        "LOADING": "Loading...",
        "LOGIN": "Login",
        "LOG_IN": "Log in",
        "LOG_OUT": "Log out",
        "MENU": "Menu",
        "NEXT": "Next",
        "NO": "No",
        "NONE": "None",
        "NO_ACCESS_HEADING": "No Access",
        "NO_ACCESS_TEXT": "Sorry, your account permissions do not allow you to access this page.",
        "NO_THANKS": "No thanks",
        "NUMBER": "Number",
        "OF": " of ",
        "PREV": "Prev",
        "PRIVACY": "Privacy",
        "REMOVE": "Remove",
        "SEARCH": "Search",
        "SEND": "Send",
        "SITE_DOWN_FOR_MAINTENANCE": "Site down for maintenance",
        "SITE_INFORMATION": "Site information",
        "TERMS_OF_USE": "Terms of use",
        "THANKS": "Thanks",
        "THANK_YOU": "Thank you",
        "VERSION": "Version",
        "YES": "Yes",
      },
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
/>
`;
